mamba-org/mamba
# Add support for configuring proxies in .condarc

Closes #151, based on #161.

This PR adds support for configuring proxies in the `.condarc` file.

Current progress:

- [x] Automated test for configuration
- [x] Automated test for proxy selection
- [x] Automated test for usage of the proxy (Might be to difficult, because an actual proxy server would be needed)

Manual tests:
- [x] Proxy without ssl_verify
- [x] Proxy with ssl_verify and cert_path
- [ ] Proxy with ssl_verify without cert_path (doesn't find certs manually added to macOS Keychain. Does OpenSSL installed by conda use it's own keychain? Seems unrelated to this PR.)
- [x] Proxy with authentication if authentication is in .condarc (Not tested yet. Should just be handled by curl)
- [ ] Proxy with authentication if authentication isn't in .condarc (Conda documentation claims it prompts the user for authentication, but in my tests it just prints an error. Mamba just fails to connect and also prints an error currently)

Open questions:
~~Should we log the proxy being used?
https://github.com/mamba-org/mamba/pull/1814/files#diff-bb45c384b0499833356c5e8fb5f6ee142bdbc44187536c1c801d2f986647beffR267~~

~~This could be very useful for debugging, however the proxy strings could contain usernames and passwords if they are configured in .condarc.~~
Password is now redacted in the log.
